java.lang.AssertionError: NoDenotation.owner

**Describe the bug**

Completions sometimes fail with **Scala 3**, yielding the following stacktrace in `metals.log`.

```
Exception in thread "pool-4-thread-3" java.lang.AssertionError: NoDenotation.owner
        at dotty.tools.dotc.core.SymDenotations$NoDenotation$.owner(SymDenotations.scala:2503)
        at scala.meta.internal.pc.ShortenedNames.tryShortenName(Signatures.scala:46)
        at scala.meta.internal.pc.ShortenedNames.tryShortenName(Signatures.scala:65)
        at scala.meta.internal.pc.Signatures$package$.loop$1(Signatures.scala:142)
        at scala.meta.internal.pc.Signatures$package$.processOwners$2(Signatures.scala:124)
        at scala.meta.internal.pc.Signatures$package$.loop$1(Signatures.scala:134)
        at scala.meta.internal.pc.Signatures$package$.$anonfun$4(Signatures.scala:170)
        at scala.collection.immutable.List.map(List.scala:246)
        at scala.meta.internal.pc.Signatures$package$.loop$1(Signatures.scala:170)
        at scala.meta.internal.pc.Signatures$package$.shortType(Signatures.scala:186)
        at scala.meta.internal.pc.SymbolPrinter.paramLabel(SymbolPrinter.scala:253)
        at scala.meta.internal.pc.SymbolPrinter.$anonfun$7(SymbolPrinter.scala:144)
        at scala.collection.immutable.List.flatMap(List.scala:293)
        at scala.meta.internal.pc.SymbolPrinter.label$2$$anonfun$1(SymbolPrinter.scala:144)
        at scala.collection.immutable.List.flatMap(List.scala:293)
        at scala.meta.internal.pc.SymbolPrinter.label$1(SymbolPrinter.scala:148)
        at scala.meta.internal.pc.SymbolPrinter.defaultMethodSignature(SymbolPrinter.scala:151)
        at scala.meta.internal.pc.SymbolPrinter.completionDetailString(SymbolPrinter.scala:96)
        at scala.meta.internal.pc.ScalaPresentationCompiler.completionItems(ScalaPresentationCompiler.scala:476)
        at scala.meta.internal.pc.ScalaPresentationCompiler.$anonfun$7(ScalaPresentationCompiler.scala:187)
        at scala.collection.immutable.List.map(List.scala:250)
        at scala.meta.internal.pc.ScalaPresentationCompiler.complete$$anonfun$1(ScalaPresentationCompiler.scala:188)
        at scala.meta.internal.pc.CompilerAccess.withSharedCompiler(CompilerAccess.scala:137)
        at scala.meta.internal.pc.CompilerAccess.$anonfun$1(CompilerAccess.scala:87)
        at scala.meta.internal.pc.CompilerAccess.onCompilerJobQueue$$anonfun$1(CompilerAccess.scala:197)
        at scala.meta.internal.pc.CompilerJobQueue$Job.run(CompilerJobQueue.scala:139)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

**To Reproduce**

- Create the following file.

```scala
// using scala 3.1.0
// using lib org.tpolecat::skunk-core:0.2.3

import skunk.syntax.all._

val a = sql""
```

- Do `scal-cli compile .` and then `code .` to open the project.
- Type a dot after `sql""` to see completions. None will appear, and a stacktrace like the above will appear in the log.

**Discussion**

As this is an **assertion** failure I assume this is actually a bug in the compiler, but who knows. It could be a problem in Metals or it could be that I failed to do something I was supposed to do in my implementation of the `sql` macro. @smarter can you comment?

--- a/mtags/src/main/scala-3/scala/meta/internal/pc/Signatures.scala
+++ b/mtags/src/main/scala-3/scala/meta/internal/pc/Signatures.scala
@@ -43,7 +43,7 @@ class ShortenedNames(
         val isOk = syms.filter(_ != NoSymbol) match
           case Nil =>
             if short.symbol.isStatic || // Java static
-              short.symbol.owner.ownersIterator.forall { s =>
+              short.symbol.maybeOwner.ownersIterator.forall { s =>
                 // ensure the symbol can be referenced in a static manner, without any instance
                 s.is(Package) || s.is(Module)
               }

